diff -ru portage_v2.2.3.orig/portage/search/pairs.cc portage_v2.2.3/portage/search/pairs.cc
--- portage_v2.2.3.orig/portage/search/pairs.cc	2020-04-27 13:29:03.000000000 -0600
+++ portage_v2.2.3/portage/search/pairs.cc	2020-06-30 09:17:20.209991049 -0600
@@ -138,8 +138,7 @@
       bool outside = false;
       pile xi(dim);
       for (int m = 0; m < dim; m++) {
-        if (x[m][i] <= cminmax[0][m]) outside = true;
-        if (x[m][i] >= cminmax[1][m]) outside = true;
+        if (x[m][i] <= cminmax[0][m] || x[m][i] >= cminmax[1][m]) outside = true;
         xi[m] = x[m][i];
       }
       if (outside) continue;
